daemon/policy.go

```diff
@@ -114,7 +122,7 @@ func (d *Daemon) UpdatePolicyEnforcement(e *endpoint.Endpoint) bool {
 			d.conf.Opts.Set(endpoint.OptionPolicy, false)
 			return false
 		}
-	} else if d.conf.EnablePolicy == endpoint.DefaultEnforcement && d.conf.IsK8sEnabled() {
+	} else if d.conf.EnablePolicy == endpoint.DefaultEnforcement && d.conf.IsK8sEnabled() && e != nil {
```

Comment:
Discussed this with Ian face to face as well. We agreed to split `UpdatePolicyEnforement` into two functions. One which takes care of endpoint specific policy enablement and one to update the global policy setting on policy changes in general.